### ISSUE TITLE
API: colWeightedMads(X) without weights gives an error

### ISSUE THREAD

**Issue by HenrikBengtsson** [HenrikBengtsson/matrixStats#24]
Argument `w` defaults to `NULL` for `col/rowWeightedMads()`, but the default gives an error, e.g.
```r
> X <- matrix(1:12, ncol=4)
> colWeightedMads(X)
Error in FUN(newX[, i], ...) :
  The number of elements in arguments 'w' and 'x' does not match: 0 != 3
```

TO DO: Detect default/missing `w` and fall back to using `colMads(X)`, cf. `colWeightedMedians()` et al.

**Comment by Dongcan-Jiang** [comment]
I tried to resolve this issue on c57dc0d04c70b34d9d238f9d9ab5f06edce00777 .

**Comment by HenrikBengtsson** [comment]
Solved in commit f52203c (same as commit f146c4e).
